Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -124,6 +120,118 @@ public override void Serialize(IIonWriter writer, object item)
             writer.StepOut();
         }
 
+        private bool DeserializeProperty(PropertyInfo property, IIonReader reader, IonType ionType, ref object deserialized)
+        {
+            if (IsReadOnlyProperty(property))
+            {
+                // property.SetValue() does not work with a readonly property.
+                // logic for handling deserializing readonly properties happens during field processing
+                // when we detect backing fields for the property.
+                return false;
+            }
+
+            deserialized = ionSerializer.Deserialize(reader, property.PropertyType, ionType);
+            if (options.IgnoreDefaults && deserialized == default)
+            {
+                return false;
+            }
+
+            return true;
+        }
+        
+        private bool DeserializeField(FieldInfo field, IIonReader reader, IonType ionType, ref object deserialized)
+        {
+            deserialized = ionSerializer.Deserialize(reader, field.FieldType, ionType);
```

Comment:
Can you move this deserialization down to just before you do the default check?

Reply:
Actually, do you have to consume the value from the reader to consume subsequent values? If so, then this is in the right place, but I wonder about the above method at line 125 since it may ignore deserialization.

Reply:
Moving the deserialization down resulted in a test failing with:
```
System.NotImplementedException: The method or operation is not implemented.
    at Amazon.IonDotnet.Internals.Text.TextScanner.SkipOverTimestamp()
```

Since this sorta thing is a possibility, I made deserialization happen at the beginning of both of these `TryDeserialize` methods.